Doc/Manual/Fortran.html

```diff
@@ -1475,19 +1317,39 @@ <H3><a name="Fortran_constructors_and_destructors">Constructors and Destructors<
 f = Foo()
 g = Foo(123)
 call f%do_something()
-call g%do_something_else()
-</code></pre>
+call g%do_something_else()</code></pre>
+</div>
+
+
+<p>
+The &ldquo;constructor&rdquo; wrapper provided by SWIG performs identically to the constructor in C++. One consequence is that C-like <code>struct</code> classes, and other classes with member data that isn&rsquo;t initialized in the constructor, will <em>not</em> have its data initialized on construction. Thus the following is undefined behavior for <code>struct Foo { int val; };</code>:
+</p>
+
+
+<div class="targetlang"><pre><code>type(Foo) :: f
+f = Foo()
+write (*,*) f%get_val()</code></pre>
 </div>
 
 
+<p>
+This is exactly analogous to the same undefined behavior in C++:
+</p>
+
+
+<div class="code"><pre><code>Foo *f = new Foo();
+cout &lt;&lt; f.val &lt;&lt; endl;</code></pre>
+</div>
+
```

Comment:
This is incorrect. `f.val` is value initialized in C++ as Foo is a POD. C++ rules are horribly complex w.r.t. initialization.

I'm lost now why the C version is not initialized. The generated code is:
```c
  result = (struct Foo *)calloc(1, sizeof(struct Foo));
```
or in C++:
```c++
  result = (Foo *)new Foo();
```
which zero initializes the struct.

Reply:
Ugh, you're right, the parens `= new Foo();` version is initialized to zero whereas `= new Foo;` is uninitialized. I'll delete this part of the documentation since clearly it doesn't apply. Good catch!